src/Serilog.Sinks.MSSqlServer/Sinks/MSSqlServer/MSSqlServerSink.cs

```diff
@@ -303,6 +303,17 @@ private string ConvertPropertiesToXmlStructure(IEnumerable<KeyValuePair<string,
             if (options.ExcludeAdditionalProperties)
                 properties = properties.Where(p => !_additionalDataColumnNames.Contains(p.Key));
 
+            if (options.PropertiesFilter != null)
+                try
+                {
+                    properties = properties.Where(p => options.PropertiesFilter(p.Key));
+                }
+                catch (Exception ex)
+                {
+                    SelfLog.WriteLine("Unable to filter properties to store due to following error: {0}", ex.Message);
```

Comment:
In general, calls to `SelfLog` try to indicate the sink in question, and the full exception info including stack trace. E.g.: https://github.com/serilog/serilog-sinks-periodicbatching/blob/dev/src/Serilog.Sinks.PeriodicBatching/Sinks/PeriodicBatching/PeriodicBatchingSink.cs#L181

Could we try:

```csharp
 SelfLog.WriteLine("Unable to filter properties to store in {0} due to following error: {1}", this, ex);
```